cocotb/types/array.py

```diff
@@ -285,42 +336,20 @@ def __setitem__(self, item, value):
                 "indexes must be ints or slices, not {}".format(type(item).__name__)
             )
 
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, type(self)):
+            if len(self) != len(other):
+                return False
+            return all(a == b for a, b in zip(self, other))
```

Comment:
What's the motivation for not just doing `self == other`?

Reply:
Ah, missed that. Originally I had `__eq__` in the ABC, but after seeing a total lack of `__eq__` in the builtin ABC, I figured that was for a good reason and moved it, but didn't simplify it.